### PR TITLE
Opgops 1935 web requirements

### PR DESCRIPTION
**Pull request by brettminnie**
* Mongodb
 * RabbitMQ
 * wkhtmlpdf
 * ssmtp

### CODE REVIEW

File: opg-kibana-1604/Dockerfile [hunk 1]
```diff
@@ -4,19 +4,19 @@ ENV KIBANA_VERSION 5.4.0
 ENV ES_SKIP_SET_KERNEL_PARAMETERS true
 
 # add our user and group first to make sure their IDs get assigned consistently
-RUN groupadd -r kibana && useradd -r -g kibana kibana
+RUN groupadd -r kibana && \
+    useradd -r -g kibana kibana && \
+    apt-get update && \
+    apt-get install -y ca-certificates curl && \
```
Comment (carinadigital) [inline opg-kibana-1604/Dockerfile]: Could installation of curl and CA be part of the next step, since it's that step that requires it

**Comment by brettminnie** [comment]
Could, probably a bit zealous on the layer reduction

File: opg-ssmtp-1604/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
+FROM registry.service.opg.digital/opg-base-1604
+
+RUN  apt update && apt install -y postfix && \
+     apt clean && apt -y autoremove && \
```
Comment (carinadigital) [inline opg-ssmtp-1604/Dockerfile]: why autoremove here?

File: opg-elasticsearch-shared-data-1604/Dockerfile [hunk 3]
```diff
@@ -7,18 +7,14 @@ ENV BEATS_DASHBOARD_VERSION=5.4.0
...
-RUN rm -rf /tmp/beats-dashboards-* && \
-   chmod -R 644 /tmp/elasticsearchshareddata
+    update-ca-certificates && \cd /tmp && \
+    wget https://artifacts.elastic.co/downloads/beats/beats-dashboards/beats-dashboards-${BEATS_DASHBOARD_VERSION}.zip && \
+    unzip beats-dashboards-${BEATS_DASHBOARD_VERSION}.zip && \
+    cd /tmp/beats-dashboards-${BEATS_DASHBOARD_VERSION} && \
+    cp -rv metricbeat/* /tmp/elasticsearchshareddata/beats && \
+    cp -rv filebeat/* /tmp/elasticsearchshareddata/beats && \
+    rm -rf /tmp/beats-dashboards-* && \
+    chmod -R 644 /tmp/elasticsearchshareddata
```
Comment (tim-mcarthur-moj) [inline opg-elasticsearch-shared-data-1604/Dockerfile]: this mode is incorrect as it is missing the x attrrib for directories.

Comment (brettminnie) [inline opg-elasticsearch-shared-data-1604/Dockerfile]: We don't need the directories to be executable though, the python script reads the files and posts them to elasticsearch @tim-mcarthur-moj

File: opg-kibana-1604/Dockerfile [hunk 4]
```diff
@@ -4,19 +4,18 @@ ENV KIBANA_VERSION 5.4.0
...
-
-# Define software versions
 ENV ELASTICSEARCH_REPO_BASE https://artifacts.elastic.co/packages/5.x/apt
-RUN echo "deb $ELASTICSEARCH_REPO_BASE stable main" > /etc/apt/sources.list.d/elasticsearch.list
 
-RUN apt-get update \
-  && apt-get install kibana=${KIBANA_VERSION}\
-  && rm -rf /var/lib/apt/lists/*
+RUN apt-get update && \
+    apt-get install -y ca-certificates curl && \
```
Comment (tim-mcarthur-moj) [inline opg-kibana-1604/Dockerfile]: are these packages really missing

File: opg-mongodb-1604/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,26 @@
+FROM registry.service.opg.digital/opg-base-1604
+
+ENV MONGODB_VERSION 3.0.15
+
+RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 7F0CEB10 && \
```
Comment (tim-mcarthur-moj) [inline opg-mongodb-1604/Dockerfile]: would it not be better to use the curl method like in kibana to fetch the key for apt so we avoid the intermittent keyserver errors

File: opg-mongodb-1604/docker/confd/templates/mongod.conf.tmpl [hunk 6]
```diff
@@ -0,0 +1,36 @@
+systemLog:
+  destination: file
+  path: "/var/log/mongodb/mongodb.log"
+  logAppend: true
+storage:
+  dbPath: /data/mongodb
+  engine: wiredTiger
+  wiredTiger:
+    engineConfig:
+      cacheSizeGB: 1
```
Comment (tim-mcarthur-moj) [inline opg-mongodb-1604/docker/confd/templates/mongod.conf.tmpl]: this should be a variable to allow performance tuning

File: opg-mongodb-1604/docker/service/mongod/run [hunk 7]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+mongod -f /etc/mongod.conf
```
Comment (tim-mcarthur-moj) [inline opg-mongodb-1604/docker/service/mongod/run]: mongo should be running as non root user

File: opg-mongodb-1604/docker/my_init/99-setup-mongo [hunk 8]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+/opt/configure_db.py &
```
Comment (tim-mcarthur-moj) [inline opg-mongodb-1604/docker/my_init/99-setup-mongo]: pushing this into the background means the server can start up without forming a cluster and the error is written belatedly to the logs, which do not seem to be sent to logstash

File: opg-kibana-1604/Dockerfile (hunk 4 above)
Comment (brettminnie) [inline opg-kibana-1604/Dockerfile]: Just tested, doesn't seem to be the case, I'll remove

File: opg-mongodb-1604/Dockerfile (hunk 5 above)
Comment (brettminnie) [inline opg-mongodb-1604/Dockerfile]: We can. latest mongodb is 3.4 as well, should we not bump it up to that now?

**Pull request merged**